packages/components/src/slot-fill/types.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import type { Component, MutableRefObject, ReactNode, RefObject } from 'react';
+
+export type DistributiveOmit< T, K extends keyof any > = T extends any
+	? Omit< T, K >
+	: never;
+
+export type SlotKey = string | symbol;
+
+export type FillProps = Record< string, any >;
+
+type SlotPropBase = {
+	/**
+	 * Slot name.
+	 */
+	name: SlotKey;
+
+	/**
+	 * props to pass from `Slot` to `Fill`.
+	 *
+	 * @default {}
+	 */
+	fillProps?: FillProps;
+};
+
+export type SlotComponentProps =
+	| ( SlotPropBase & {
+			/**
+			 * By default, events will bubble to their parents on the DOM hierarchy (native event bubbling).
+			 * If set to true, events will bubble to their virtual parent in the React elements hierarchy instead,
+			 * also accept an optional `className`, `id`, etc.  to add to the slot container.
+			 */
+			bubblesVirtually: true;
+
+			/**
+			 * A function that returns nodes to be rendered.
+			 * Not supported when `bubblesVirtually` is true.
+			 */
+			children?: never;
+
+			/**
+			 * className.
+			 * Not supported when `bubblesVirtually` is true.
+			 */
+			className?: string | undefined;
```

Comment:
Is the `| undefined` redundant?

```suggestion
			/**
			 * A function that returns nodes to be rendered.
			 * Not supported when `bubblesVirtually` is true.
			 */
			children?: never;

			/**
			 * className.
			 * Not supported when `bubblesVirtually` is true.
			 */
			className?: string;
```